### ISSUE TITLE
error=build for binary does not contain a main function

### ISSUE THREAD

**Issue by greenpau** [goreleaser/goreleaser-action#317]
I have a vanilla [config](https://github.com/greenpau/ovn_exporter/blob/250811bb207eb33e9d81cdd15b5cfa1a5043e99a/.github/workflows/build.yml#L1-L38). The manually triggered action results in the following error:

```
release failed after 5.33s error=build for ovn_exporter does not contain a main function
```

The failed run: [greenpau/ovn_exporter/runs/4358394650](https://github.com/greenpau/ovn_exporter/runs/4358394650?check_suite_focus=true)

How do I troubleshoot the failure?

**Comment by greenpau** [comment]
Running `$ goreleaser build --snapshot --rm-dist` locally and getting the same error:

```
      • building                  binary=/home/greenpau/dev/go/src/github.com/greenpau/ovn_exporter/dist/ovn_exporter_linux_amd64/ovn_exporter
   ⨯ build failed after 0.11s error=build for ovn_exporter does not contain a main function
```

The `dist` directory contains a single `dist/config.yaml`. The `ovn_exporter_linux_amd64/ovn_exporter` are not there.

Perhaps, there has to be a step `make build` type of step? 🤔

**Comment by caarlos0** [comment]
> Perhaps, there has to be a step `make build` type of step? 🤔

no, probably you haven't created a `.goreleaser.yml` and your main function is not on the root folder.

Please refer to the documentation: https://goreleaser.com/quick-start/ and https://goreleaser.com/customization/build/

**Comment by greenpau** [comment]
> no, probably you haven't created a .goreleaser.yml and your main function is not on the root folder.

@caarlos0 , 👍 my `main` function is not in the root. I do have [.goreleaser.yml](https://github.com/greenpau/ovn_exporter/blob/main/.goreleaser.yml) though.

**Comment by caarlos0** [comment]
you are not informing the `main` property in your build config, thats likely the issue

**Comment by caarlos0** [comment]
Improved the error msg a bit: https://github.com/goreleaser/goreleaser/pull/2709

**Comment by caarlos0** [comment]
so, was that it? can we close this?

**Comment by greenpau** [comment]
>so, was that it? can we close this?

@caarlos0 , thank you for your help! 👍 

I am all good. Released [v1.0.4](https://github.com/greenpau/ovn_exporter/releases/tag/v1.0.4). Had to do the following changes:

* [.github/workflows/build.yml](https://github.com/greenpau/ovn_exporter/blob/2db77a076f4592ad4784d475802fe616aa38a2fa/.github/workflows/build.yml#L33)
* [.goreleaser.yml](https://github.com/greenpau/ovn_exporter/blob/2db77a076f4592ad4784d475802fe616aa38a2fa/.goreleaser.yml#L6-L28)

**Comment by caarlos0** [comment]
Nice! Glad it worked out!

Will close this then, but feel free to reopen/comment/open another issue if needed.

We also have a channel in the gopher slack and on discord, links to all https://goreleaser.com/links/
